02-peer-protocol.md

```diff
@@ -492,6 +493,58 @@ to broadcast the commitment transaction to get his funds back and open a new
 channel. To avoid this, the funder should ensure the funding transaction
 confirms in the next 2016 blocks.
 
+## Channel Quiescence
+
+Various fundamental changes, in particular protocol upgrades, are
+easiest on channels where both commitment transactions match, and no
+pending updates are in flight.  We define a protocol to quiesce the
+channel by indicating that "SomeThing Fundamental is Underway".
+
+### `stfu`
+
+1. type: 2 (`stfu`)
+2. data:
+    * [`channel_id`:`channel_id`]
+    * [`u8`:`initiator`]
+
+### Requirements
+
+The sender of `stfu`:
+  - MUST NOT send `stfu` unless `option_quiesce` is negotiated.
+  - MUST NOT send `stfu` if any of the sender's htlc additions, htlc removals
+    or fee updates are pending for either peer.
```

Comment:
Good point.  This refers to the state machine, where changes are referred to as pending.

Concretely, this is not allowed if you have sent update_* messages, without a commitment_signed.  Is that clearer?

Reply:
Yes that would have been much clearer. But maybe not necessary at that point of the document if it is clear within BOLT2 that the word pending always refers to the state machine and not to in flight HTLCs. If not I suggest to introduce such a clarification the first time the word pending is being used in BOLT 2. Also as said I eventually understood the meaning of pending so the issue seems to be not that big.

Reply:
I am having trouble understanding what "pending" means here.

In this flow, is the add considered pending?:
```
--update_add_htlc-->
--commit_sig------->
```

If so, is it only considered non-pending at the end of this flow?:
```
--update_add_htlc-->
--commit_sig------->
<--revoke_and_ack--
<--commit_sig------
--revoke_and_ack-->
```

Reply:
> Agree that "pending" is confusing here. I think it would be more clear to write:
> 
>     * MUST NOT send `stfu` if any of the sender's htlc additions, htlc removals
>       or fee updates have not been counter signed or failed by our peer.

I agree. I think it's still vague.